## Importing numpy and pandas libraries

In [1]:
import numpy as np
import pandas as pd

## Importing the datasets-2006 and 2007

In [2]:
dataset06 = pd.read_csv("../Coursework R/2006.csv")
dataset07 = pd.read_csv("../Coursework R/2007.csv")

In [3]:
#Checking column names in dataset06
dataset06.columns

Index(['Year', 'Month', 'DayofMonth', 'DayOfWeek', 'DepTime', 'CRSDepTime',
       'ArrTime', 'CRSArrTime', 'UniqueCarrier', 'FlightNum', 'TailNum',
       'ActualElapsedTime', 'CRSElapsedTime', 'AirTime', 'ArrDelay',
       'DepDelay', 'Origin', 'Dest', 'Distance', 'TaxiIn', 'TaxiOut',
       'Cancelled', 'CancellationCode', 'Diverted', 'CarrierDelay',
       'WeatherDelay', 'NASDelay', 'SecurityDelay', 'LateAircraftDelay'],
      dtype='object')

In [4]:
#Checking column names in dataset07
dataset07.columns

Index(['Year', 'Month', 'DayofMonth', 'DayOfWeek', 'DepTime', 'CRSDepTime',
       'ArrTime', 'CRSArrTime', 'UniqueCarrier', 'FlightNum', 'TailNum',
       'ActualElapsedTime', 'CRSElapsedTime', 'AirTime', 'ArrDelay',
       'DepDelay', 'Origin', 'Dest', 'Distance', 'TaxiIn', 'TaxiOut',
       'Cancelled', 'CancellationCode', 'Diverted', 'CarrierDelay',
       'WeatherDelay', 'NASDelay', 'SecurityDelay', 'LateAircraftDelay'],
      dtype='object')

-Since the dataset06 and dataset07 contain the same columns we combine them to form a combined dataset named datacomb.

In [5]:
datacomb = pd.concat([dataset06,dataset07])
datacomb.head()

Year  Month  DayofMonth  DayOfWeek  DepTime  CRSDepTime  ArrTime  \
0  2006      1          11          3    743.0         745   1024.0   
1  2006      1          11          3   1053.0        1053   1313.0   
2  2006      1          11          3   1915.0        1915   2110.0   
3  2006      1          11          3   1753.0        1755   1925.0   
4  2006      1          11          3    824.0         832   1015.0   

   CRSArrTime UniqueCarrier  FlightNum  ... TaxiIn  TaxiOut  Cancelled  \
0        1018            US        343  ...     45       13          0   
1        1318            US        613  ...     27       19          0   
2        2133            US        617  ...      4       11          0   
3        1933            US        300  ...     16       10          0   
4        1015            US        765  ...     27       12          0   

   CancellationCode  Diverted  CarrierDelay WeatherDelay NASDelay  \
0               NaN         0             0            0        0   
1               NaN         0             0            0        0   
2               NaN         0             0            0        0   
3               NaN         0             0            0        0   
4               NaN         0             0            0        0   

   SecurityDelay  LateAircraftDelay  
0              0                  0  
1              0                  0  
2              0                  0  
3              0                  0  
4              0                  0  

[5 rows x 29 columns]

## Removing duplicates from the combined dataset

In [6]:
datacomb=datacomb.drop_duplicates()

## Exploring the combined dataset

In [7]:
#The number of columns and rows in datacomb is identified
datacomb.shape

(14595103, 29)

In [8]:
#Checking the column names in datacomb
datacomb.columns

Index(['Year', 'Month', 'DayofMonth', 'DayOfWeek', 'DepTime', 'CRSDepTime',
       'ArrTime', 'CRSArrTime', 'UniqueCarrier', 'FlightNum', 'TailNum',
       'ActualElapsedTime', 'CRSElapsedTime', 'AirTime', 'ArrDelay',
       'DepDelay', 'Origin', 'Dest', 'Distance', 'TaxiIn', 'TaxiOut',
       'Cancelled', 'CancellationCode', 'Diverted', 'CarrierDelay',
       'WeatherDelay', 'NASDelay', 'SecurityDelay', 'LateAircraftDelay'],
      dtype='object')

In [9]:
#Checking the datatypes of the column variables
datacomb.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 14595103 entries, 0 to 7453214
Data columns (total 29 columns):
 #   Column             Dtype  
---  ------             -----  
 0   Year               int64  
 1   Month              int64  
 2   DayofMonth         int64  
 3   DayOfWeek          int64  
 4   DepTime            float64
 5   CRSDepTime         int64  
 6   ArrTime            float64
 7   CRSArrTime         int64  
 8   UniqueCarrier      object 
 9   FlightNum          int64  
 10  TailNum            object 
 11  ActualElapsedTime  float64
 12  CRSElapsedTime     float64
 13  AirTime            float64
 14  ArrDelay           float64
 15  DepDelay           float64
 16  Origin             object 
 17  Dest               object 
 18  Distance           int64  
 19  TaxiIn             int64  
 20  TaxiOut            int64  
 21  Cancelled          int64  
 22  CancellationCode   object 
 23  Diverted           int64  
 24  CarrierDelay       int64  
 25  WeatherDelay     

## Checking for missing values

In [10]:
pd.isnull(datacomb).sum(axis=0)

Year                        0
Month                       0
DayofMonth                  0
DayOfWeek                   0
DepTime                282682
CRSDepTime                  0
ArrTime                316046
CRSArrTime                  0
UniqueCarrier               0
FlightNum                   0
TailNum                    22
ActualElapsedTime      316046
CRSElapsedTime            998
AirTime                316046
ArrDelay               316046
DepDelay               282682
Origin                      0
Dest                        0
Distance                    0
TaxiIn                      0
TaxiOut                     0
Cancelled                   0
CancellationCode     14312420
Diverted                    0
CarrierDelay                0
WeatherDelay                0
NASDelay                    0
SecurityDelay               0
LateAircraftDelay           0
dtype: int64

-Missing values will be removed where needed when answering the questions

## Importing the other csv files

In [11]:
airports = pd.read_csv("../Coursework R/airports.csv")
planedata = pd.read_csv("../Coursework R/plane-data.csv")

## Removing duplicates from these datasets 

In [12]:
airports=airports.drop_duplicates()
planedata=planedata.drop_duplicates()

## Checking for missing values for these datasets

In [13]:
#checking missing values in 'airports' dataset
pd.isnull(airports).sum(axis = 0)

iata        0
airport     0
city       12
state      12
country     0
lat         0
long        0
dtype: int64

12 missing values under city and 12 missing values under state columns in 'airports' dataset

In [14]:
airports = airports[pd.notnull(airports['city'])]

In [15]:
pd.isnull(airports).sum(axis = 0)

iata       0
airport    0
city       0
state      0
country    0
lat        0
long       0
dtype: int64

All the missing values in 'airports' dataset is removed!

In [16]:
# Checking missing values in 'planedata'
pd.isnull(planedata).sum(axis = 0)

tailnum            0
type             549
manufacturer     549
issue_date       549
model            549
status           549
aircraft_type    549
engine_type      549
year             549
dtype: int64

There are 549 missing values in all columns of 'planedata' dataset except in tailnum column!

In [17]:
planedata = planedata[pd.notnull(planedata['year'])]

In [18]:
pd.isnull(planedata).sum(axis = 0)

tailnum          0
type             0
manufacturer     0
issue_date       0
model            0
status           0
aircraft_type    0
engine_type      0
year             0
dtype: int64

All missing values in 'planedata' dataset is removed!

## Saving the cleaned datasets

In [20]:
datacomb.to_csv("../Coursework/datacomb.csv")
airports.to_csv("../Coursework/airports_c.csv")
planedata.to_csv("../Coursework/planedata_c.csv")